src/main/java/org/zalando/nakadi/service/CursorsService.java

```diff
@@ -106,15 +99,14 @@ public CursorsService(final ZooKeeperHolder zkHolder,
 
         return cursors.stream()
                 .map(cursor -> {
-                    final EventTypePartition etPartition = new EventTypePartition(cursor.getEventType(),
+                    final EventTypePartition etPartition = new EventTypePartition(cursor.getTimeline().getEventType(),
                             cursor.getPartition());
-                    final Boolean committed = partitionCommits.get(etPartition).next();
-                    return new CursorCommitResult(cursor, committed);
+                    return partitionCommits.get(etPartition).next();
                 })
                 .collect(Collectors.toList());
     }
 
-    private void validateStreamId(final List<SubscriptionCursor> cursors, final String streamId,
+    private void validateStreamId(final List<NakadiCursor> cursors, final String streamId,
```

Comment:
On line 133:
partitionSession = getPartitionSession(subscriptionId, eventType.getTopic(), cursor);
 you take topic from event type. I think it does not make much sense.